packages/react/src/hooks/useVoiceAssistant.ts

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { ParticipantKind, Track } from 'livekit-client';
+import type { RemoteParticipant } from 'livekit-client';
+import type { ReceivedTranscriptionSegment, TrackReference } from '@livekit/components-core';
+import { useRemoteParticipants } from './useRemoteParticipants';
+import { useParticipantTracks } from './useParticipantTracks';
+import { useTrackTranscription } from './useTrackTranscription';
+import { useParticipantInfo } from './useParticipantInfo';
+import { useConnectionState } from './useConnectionStatus';
+
+/**
+ * @alpha
+ */
+export type VoiceAssistantState = 'offline' | 'connecting' | 'listening' | 'thinking' | 'speaking';
+
+/**
+ * @alpha
+ */
+export interface VoiceAssistant {
+  agent: RemoteParticipant | undefined;
+  state: VoiceAssistantState;
+  audioTrack: TrackReference | undefined;
+  agentTranscriptions: ReceivedTranscriptionSegment[];
+}
+
+/**
+ * @alpha
+ *
+ * This hook looks for the first agent-participant in the room.
+ * It assumes that the agent participant is based on the LiveKit VoiceAssistant API and
+ * returns the most commonly used state vars when interacting with a VoiceAssistant.
+ */
+export function useVoiceAssistant(): VoiceAssistant {
+  const agent = useRemoteParticipants().find((p) => p.kind === ParticipantKind.AGENT);
+  const audioTrack = useParticipantTracks([Track.Source.Microphone], agent?.identity)[0];
```

Comment:
if it hasn't been published yet, would this be accessing an invalid element of array?

Reply:
audiotrack would be `undefined`, but not raising an exception

Reply:
good to know! some languages would raise.